package T5;

public class Except {
	
	public int div(int a,int b) throws Exception //throws Exception
	{
		//return a/b;
		/*try{
			return a/b;
			}
		catch (Exception e){
			//e.printStackTrace();
			System.out.println(e.toString());
			throw e;
			//return 1;
			}*/
			if (b<0) throw new JennysException("Blah");
		}
	
	public int fn1(int a,int b) throws Exception{
		return div(a,b);
		
	}
}

class JennysException extends Exception{
	JennysException(String str){
		super(str);
	}
}
class ExceptTest{
	public static void main(String[] args) throws Exception{
		Except ex = new Except();
		ex.fn1(10,0);
		//ex.fn1(10,0);
		/*try{
			ex.fn1(10,0);
		}
		catch (ArithmeticException e){
			//System.out.println(e.toString());
			e.printStackTrace();
		}
		catch (Exception e){
			System.out.println(e.toString());
		}*/
		System.out.println("Finsh!");
	}
}
